src/server/main_service.cc

```diff
@@ -2072,7 +2072,9 @@ GlobalState Service::GetGlobalState() const {
 
 void Service::ConfigureHttpHandlers(util::HttpListenerBase* base, bool is_privileged) {
   // We skip authentication on privileged listener if the flag admin_nopass is set
-  const bool should_skip_auth = is_privileged && !RequirePrivilegedAuth();
+  // We also skip authentication if requirepass is empty
+  const bool should_skip_auth =
+      (is_privileged && !RequirePrivilegedAuth()) || GetPassword().empty();
```

Comment:
Nit: you could add a `string password = GetPassword()` variable instead of calling the function twice. On the other hand, it's not a hot path at all, so your call if you think it's nicer.

Reply:
I thought of this as well but this is during the init and it's only called at maximum 2 times :stuck_out_tongue: